homeassistant/components/maxcube/binary_sensor.py

```diff
@@ -26,13 +23,12 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class MaxCubeShutter(BinarySensorEntity):
     """Representation of a MAX! Cube Binary Sensor device."""
 
-    def __init__(self, handler, name, rf_address):
+    def __init__(self, handler, device):
         """Initialize MAX! Cube BinarySensorEntity."""
-        self._name = name
-        self._sensor_type = DEVICE_CLASS_WINDOW
-        self._rf_address = rf_address
+        room = handler.cube.room_by_id(device.room_id)
```

Comment:
Is this just a lookup of cached data or is it doing I/O?

Reply:
handler.update() fetches all the state information. All getters returns the cached information.